v1.20.0: begin changelog (#575)
fusedio/fused-docs: docs/python-sdk/changelog.md

# Changelog

## v1.20.0 (2025-06-__)

**[`fused-py`](/python-sdk/)**

New Features:
- Running a UDF with engine `local` will cache similarly to how it would when running on `remote`.

> - Large (batch) jobs can be accepted in a Pending state if they cannot start immediately.

Reviewer: It's not clear to me what this "can be accepted" means. Accepted in which context?

Author: reworded it to try to make it clearer

- UDFs can now accept `**kwargs` parameters, which will always be passed in as strings.
- Upgraded DuckDB to 1.3.0.
- `@fused.cache` has new `cache_verbose` options. If set to `True`, Print a message when a cached result is returned

> - `@fused.cache` has new `cache_reset` option. The existing `reset` parameter is deprecated.

Reviewer: ```suggestion
- `@fused.cache` renamed the `reset` parameter to `cache_reset`. The existing `reset` parameter is deprecated.
```

- Some file listing APIs like `fused.api.list` will work for public buckets when on free accounts.

> - `fused.load` accepts a parameter for controlling importing UDF globals. Also, when globals cannot be imported, a warning is emitted instead of an exception.

Reviewer 2: ```suggestion
- `fused.load` accepts `import_globals` (default `True`) for controlling importing UDF globals. Also, when globals cannot be imported, a warning is emitted instead of an exception.
```

Bug Fixes:
- Clarified login-needed message in Fused Apps.
- Fixed bugs with `@fused.cache` results not being ready.
- Fixed bugs with `@fused.cache` not detecting changes in the cached function.
- Loading a UDF from a file will autodetect the UDF function name.
- Fixed bugs with returning GeoDataFrames that do not contain geometry.
- Fixed calling `to_fused` on an app.

**[Workbench](/workbench/)**

- A message will appear above the UDF body when parameters are set in the UDF list.
- A lock icon will be shown next to read-only UDFs and Apps in Workbench.
- When pushing UDFs to GitHub, the preview image will be pushed to a public URL so that the README in GitHub is rendered correctly.
- When pushing UDFs to GitHub, Workbench will assign the PR to you if possible.
- It is now possible to delete UDFs and Apps directly from the catalog.

> - Added a "Reload Collection" button.

Reviewer 2: ```suggestion
- Added a "Reload Collection" button. This pulls all latest version of UDF currently in your Collection.
```
